controllers/storagecluster/storageclasses.go

```diff
@@ -287,8 +287,12 @@ func (r *StorageClusterReconciler) newStorageClassConfigurations(initData *ocsv1
 	if initData.Spec.Encryption.StorageClass && initData.Spec.Encryption.KeyManagementService.Enable {
 		kmsConfig, err := getKMSConfigMap(KMSConfigMapName, initData, r.Client)
 		if err == nil && kmsConfig != nil {
-			serviceName := kmsConfig.Data["KMS_SERVICE_NAME"]
-			ret = append(ret, newEncryptedCephBlockPoolStorageClassConfiguration(initData, serviceName))
+			authMethod, found := kmsConfig.Data["VAULT_AUTH_METHOD"]
+			// for 4.10, skipping SC creation for vault SA based kms encryption
+			if !(found && authMethod == VaultSAAuthMethod) {
```

Comment:
```suggestion
			if found && authMethod != VaultSAAuthMethod {
```

Reply:
We are supporting multiple KMS services:
    1. Vault
    2. IBM Hpcs
1.  The condition should be if kms service is vault and auth method is not Kubernetes then continue.
2. Also, if KMS service is not Vault then also continue.
 
Just to reduce if condition i used this, 

    !(found && authMethod == VaultSAAuthMethod) always true for non Vault  kms services. because other kms service dont have "VAULT_AUTH_METHOD".
    For Vault it will be true only authMethod is not equal to Kubernetes

Reply:
i cant use found && authMethod != VaultSAAuthMethod because found always false for non Vault kms service.